C/tests/ParserTest_0000_0999.cpp

```diff
@@ -2015,102 +2015,464 @@ void ParserTest::case0279()
+                              IdentifierDeclarator,
+                              CompoundStatement,
+                              ReturnStatement,
+                              AddExpression,
+                              IdentifierName,
+                              IdentifierName }));
 }
 
 void ParserTest::case0285()
 {
-
+    parse("int x ( y , z ) int y ; int z ( int ) ; { return y + z ( y ) ; }",
+          Expectation().AST({ TranslationUnit,
+                              FunctionDefinition,
+                              BuiltinTypeSpecifier,
+                              FunctionDeclarator,
+                              IdentifierDeclarator,
+                              ParameterSuffix,
+                              ParameterDeclaration,
+                              TypedefName,
+                              AbstractDeclarator,
+                              ParameterDeclaration,
+                              TypedefName,
+                              AbstractDeclarator,
+                              ExtKR_ParameterDeclaration,
+                              BuiltinTypeSpecifier,
+                              IdentifierDeclarator,
+                              ExtKR_ParameterDeclaration,
+                              BuiltinTypeSpecifier,
+                              FunctionDeclarator,
+                              IdentifierDeclarator,
+                              ParameterSuffix,
+                              ParameterDeclaration,
+                              BuiltinTypeSpecifier,
+                              AbstractDeclarator,
+                              CompoundStatement,
+                              ReturnStatement,
+                              AddExpression,
+                              IdentifierName,
+                              CallExpression,
+                              IdentifierName,
+                              IdentifierName }));
 }
 
 void ParserTest::case0286()
 {
-
+    parse("int x ( y ) struct t y ; { return y . z ; }",
+          Expectation().AST({ TranslationUnit,
+                              FunctionDefinition,
+                              BuiltinTypeSpecifier,
+                              FunctionDeclarator,
+                              IdentifierDeclarator,
+                              ParameterSuffix,
+                              ParameterDeclaration,
+                              TypedefName,
+                              AbstractDeclarator,
+                              ExtKR_ParameterDeclaration,
+                              StructTypeSpecifier,
+                              IdentifierDeclarator,
+                              CompoundStatement,
+                              ReturnStatement,
+                              DirectMemberAccessExpression,
+                              IdentifierName,
+                              IdentifierName }));
 }
 
 void ParserTest::case0287()
 {
-
+    parse("int x ( y ) t y ; { return y . z ; }",
```

Comment:
In the tests I only use `x`, `y`, `z`, or `w`. If you use `t` there's an implicit suggestion that the parser can differentiate an identifier that names a type or a variable, which isn't the case. See my comment above when you say _"If the function does not have any typed parameter, it is treated"_